manta-util/src/codec.rs

```diff
@@ -587,51 +587,65 @@ impl<T> Encode for PhantomData<T> {
     }
 }
 
-impl Encode for u8 {
+impl Encode for bool {
     #[inline]
-    fn encode<W>(&self, mut writer: W) -> Result<(), W::Error>
+    fn encode<W>(&self, writer: W) -> Result<(), W::Error>
     where
         W: Write,
     {
-        writer.write_ref(&[*self])?;
-        Ok(())
+        (*self as u8).encode(writer)
     }
 }
 
-impl Encode for u16 {
+impl Encode for u8 {
     #[inline]
     fn encode<W>(&self, mut writer: W) -> Result<(), W::Error>
     where
         W: Write,
     {
-        writer.write_ref(&self.to_le_bytes())?;
+        writer.write_ref(&[*self])?;
         Ok(())
     }
 }
 
-impl Encode for u32 {
-    #[inline]
-    fn encode<W>(&self, mut writer: W) -> Result<(), W::Error>
-    where
-        W: Write,
-    {
-        writer.write_ref(&self.to_le_bytes())?;
-        Ok(())
+/// Defines an [`Encode`] implemention for the given integer type `$type`.
+macro_rules! encode_int {
+    ($($type:tt),* $(,)?) => {
+        $(
+            impl Encode for $type {
+                #[inline]
+                fn encode<W>(&self, mut writer: W) -> Result<(), W::Error>
+                where
+                    W: Write,
+                {
+                    writer.write_ref(&self.to_le_bytes())?;
+                    Ok(())
+                }
+            }
+        )*
     }
 }
 
-impl Encode for u64 {
+encode_int!(u16, i16, u32, i32, u64, i64, u128, i128);
```

Comment:
Why does u8 have a separate implementation? And where's i8?

Reply:
It's separate because it's the canonical data type for raw data and the other integer types use it's implementation when defining their own so it cannot use the macro. I'll add an `i8`.

Reply:
Actually, it doesn't have to be separate. I fixed it.